### ISSUE TITLE
Add support for displaying song covers from an external API

### ISSUE THREAD

**Issue by SlvrEagle23** [AzuraCast/AzuraCast#102]

**Comment by SlvrEagle23** [comment]
It turns out this is actually a more complicated issue than it sounds.

The discussion on this StackOverflow question sheds some light on the difficulty:
https://stackoverflow.com/questions/7543574/is-there-any-free-unlimited-album-artwork-search-api-service

In particular, a majority of the cover art APIs require that an API key be obtained for the application; because AzuraCast is a redistributable application, we can't package an API key because it would then be public on our repository and anyone could use it. We could ask users to generate their own API keys, but even then another problem occurs: Last.fm (the largest API in question) says that cover art from their API cannot be included in any redistributable application due to licensing issues relating to ownership of the album art.

Long story short, it's a surprisingly complicated problem and it's something that AzuraCast wasn't really built to solve all that well.

**Comment by Vaalyn** [comment]
@SlvrEagle23 Just stumbled upon this thread and did some research. I can't find any information on the [terms of service](https://www.last.fm/api/tos) of the Last.fm API that says that we can't distribute software which offers a Last.fm API integration that the user has to provide an API key with. Just a note that the user has to get a written agreement from Last.fm for their page which is not something that we can take care of for the user.

> You agree to only give public access to pages using Last.fm's web services that have been previously approved by Last.fm in writing.

Is there anything I missed here?

**Comment by ANSASERVERS** [comment]
The user should provide their own API key - You just need to provide the ability to enter it and do the back end coding to make it work when the key is entered.

**Comment by SlvrEagle23** [comment]
Support for external album art retrieval has been added in the latest stable release.
